pkg/util/math/math.go

```diff
@@ -36,3 +38,11 @@ func Min64(a, b int64) int64 {
 	}
 	return b
 }
+
+// MinDuration returns the minimum of two time.Durations
+func MinDuration(a, b time.Duration) time.Duration {
```

Comment:
This solution works fine, but I think that's a case where generics might also make sense.